File: leet_code/linked_lists/reverse_list.js
const reverseList = head => {
  if (!head) return null;
  if (!head.next) return head;

  let finalHead = head;
  let nextNode = head.next;

  while (nextNode) {
    head.next = nextNode.next;
    nextNode.next = finalHead;
    finalHead = nextNode;
    nextNode = head.next;
  }
  return finalHead;
};

File: leet_code/linked_lists/reverse_list.rb
def reverse_list(head)
    return nil if !head
    return head if !head.next

    final_head = head
    next_node = head.next

    while next_node
        head.next = next_node.next
        next_node.next = final_head
        final_head = next_node
        next_node = head.next
    end
    final_head
end
